### PR TITLE
Added functionality for change status of card and congratulation

### PR DESCRIPTION
**Pull request by RomanKlishch**
- added method for change status of congratulations;
- added method for change status of card;
- added tests

### CODE REVIEW

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java [hunk 1]
```diff
@@ -31,10 +32,14 @@
...
     private static final String DELETE_BY_CARD_ID = "DELETE FROM cards WHERE card_id=? and user_id=?";
     private static final String FIND_LINKS_BY_CARD_ID = "SELECT link FROM links l LEFT JOIN congratulations cg ON cg.congratulation_id = l.congratulation_id where card_id=? and (type_id = 2 OR type_id = 3) and user_id =?";
+    private static final String CHANGE_STATUS_OF_CARD_BY_ID = "UPDATE cards SET status_id = ? where card_id = ?";
+
 
     private static final CardRowMapper CARD_ROW_MAPPER = new CardRowMapper();
     private static final CardAndCongratulationRowMapper CARD_AND_CONGRATULATION_ROW_MAPPER = new CardAndCongratulationRowMapper();
     private final PropertyReader propertyReader = ServiceLocator.getBean("PropertyReader");
+    private final CongratulationDao congratulationDao = ServiceLocator.getBean("JdbcCongratulationDao");
+
```
Comment (Yuliia-Sh) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCardDao.java:42]: Why to not put into constructor?

File: src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class FinishCardServlet extends HttpServlet {
+    private CardService cardService = ServiceLocator.getBean("DefaultCardService");
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+        long cardId = Long.parseLong("1");
+        cardService.changeCardStatus(Status.ISOVER,cardId);
+    }
```
Comment (Yuliia-Sh) [inline src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java:17]: get request shouldn't change data

Comment (Yuliia-Sh) [reply src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java:17]: Long.parseLong("1"); - is it test?

File: src/test/java/com/greetingcard/dao/jdbc/JdbcCardDaoITest.java [hunk 3]
```diff
@@ -175,5 +178,15 @@ void deleteCardById() throws IOException {
...
     }
 
+    @Test
+    @DisplayName("Change status of card to ISOVER")
+    void changeStatusCardById() {
+        //when
+        jdbcCardDao.changeStatusCardById(Status.ISOVER, 1);
+        //then
+        Card card = jdbcCardDao.getCardAndCongratulationByCardId(1, 1);
+        assertEquals(Status.ISOVER, card.getStatus());
```
Comment (Yuliia-Sh) [inline src/test/java/com/greetingcard/dao/jdbc/JdbcCardDaoITest.java:188]: for tests were added new methods? but they also should be covered by tests then

**Review by Yuliia-Sh — COMMENTED** [review]
could you please look at my comments

File: src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import com.greetingcard.service.CardService;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class FinishCardServlet extends HttpServlet {
+    private CardService cardService = ServiceLocator.getBean("DefaultCardService");
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
```
Comment (AlexKievDeveloper) [inline src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java:14]: I think better use the whole words: request and response

File: src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        long cardId = Long.parseLong("1");
+        cardService.changeCardStatus(Status.ISOVER,cardId);
```
Comment (AlexKievDeveloper) [inline src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java:16]: Will we always use 1 as cardId?

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java [hunk 6]
```diff
@@ -97,17 +102,44 @@ public void leaveByCardId(long cardId, long userId) {
...
         } catch (SQLException e) {
-            log.error("Exception while deleting congratulations by - {}" , cardId, e);
-            throw new RuntimeException("Exception while deleting congratulations " , e);
+            log.error("Exception while deleting congratulations by - {}", cardId, e);
+            throw new RuntimeException("Exception while deleting congratulations ", e);
+        }
+    }
+
+    @Override
+    public List<Congratulation> findCongratulationsByCardId(long cardId) {
```
Comment (AlexKievDeveloper) [inline src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java:120]: Where you use this method?

**Review by AlexKievDeveloper — APPROVED** [review]

File: src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java (hunk 2 above)
Comment (RomanKlishch) [reply src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java:17]: I am foget delete this row

File: src/test/java/com/greetingcard/dao/jdbc/JdbcCardDaoITest.java (hunk 3 above)
Comment (RomanKlishch) [reply src/test/java/com/greetingcard/dao/jdbc/JdbcCardDaoITest.java:188]: They have tests. I added tests for this method several days ago in another task

File: src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java (hunk 6 above)
Comment (RomanKlishch) [reply src/main/java/com/greetingcard/dao/jdbc/JdbcCongratulationDao.java:120]: I used this method for another implementation of may task

**Review by AlexKievDeveloper — DISMISSED** [review]

File: src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import com.greetingcard.service.CardService;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class FinishCardServlet extends HttpServlet {
+    private CardService cardService = ServiceLocator.getBean("DefaultCardService");
+    
+}
```
Comment (Yuliia-Sh) [inline src/main/java/com/greetingcard/web/servlet/card/FinishCardServlet.java:14]: Why empty class added?

**Review by Yuliia-Sh — APPROVED** [review]

**Review by OleksandrGubariev — APPROVED** [review]

**Review by Yuliia-Sh — APPROVED** [review]

**Review by AlexKievDeveloper — APPROVED** [review]

**Review by OleksandrGubariev — APPROVED** [review]

**Pull request merged**